Titles to Keywords Rewrite

latenitefilms opened the pull request:
- Changed how "Titles to Keywords" functions internally. We now basically get the timeline position of every title, then compare it to every clip on the timeline. This means the connection point of clips is no longer relevant.
- Closes #3094

latenitefilms committed:
Improvements

- Changed how "Titles to Keywords" functions internally. We now basically get the timeline position of every title, then compare it to every clip on the timeline. This means the connection point of clips is no longer relevant.
- Closes #3094

latenitefilms committed:
Fixed @stickler-ci complaints

latenitefilms committed:
Fixed documentation

randomeizer commented:
This might be useful:

https://github.com/torch/rational

randomeizer commented:
Possibly this also:

https://exercism.org/tracks/lua/exercises/rational-numbers/solutions

latenitefilms committed:
Improvements

- Removed `cp.apple.fcpxml.timeStringToSeconds()` and `cp.apple.fcpxml.numberToTimeString()`
- Added `cp.apple.fcpxml.time` helper.
- Revised Titles to Keywords Toolbox to use the new `cp.apple.fcpxml.time` helper.
- Getting closer... but still not working in all scenarios. Something is broken in the logic.
- Work in progress.

latenitefilms committed:
Improvements

- Renamed variables to make the code clearer.
- Fixed bug with how gap positions are calculated.
- Tested and working with 25fps clips in a 25fps timeline.

latenitefilms committed:
Fixed @stickler-ci errors

latenitefilms committed:
Work in progress

- Detecting the title position on the timeline is now correct. Everything else is broken/wrong.

latenitefilms committed:
Bug Fixes

- Clips on Secondary Storylines now work correctly.

latenitefilms committed:
Removed unnecessary code

latenitefilms committed:
Cleanup

- Removed debugging code.
- Simplified and reformatted code.
